packages/from-query/src/generate.ts

```diff
@@ -129,17 +149,32 @@ export default (OPTIONS: IFromQueryOptions): (ir: IOperation) => string => {
                 interfaceName: name
               }
             }
+
             return name;
           });
-          return printField(
-            this.generateSelectionName(selection),
-            selection.typeDefinition,
-            selection
-          );
+
+          const selectionHasDirectives = hasDirectives(selection.directives);
+          const fragmentsHaveDirectives = !!selection.fragments.some(frag => hasDirectives(frag.directives))
```

Comment:
An interesting thing to bring up is the `defer` directive. what should happen there? Should it transform a field from `T -> T | null`? Should it generate two types one for the deferred and one for the final result and then OR them?

What about custom directives? This was something I gave a little thought to but didn't come up with a consensus

Reply:
I think re: custom directives it was an idea to allow the user to provide a "transform" function of sorts. but I don't exactly know what that looks like